src/irmin-pack/atomic_write.ml

```diff
@@ -220,6 +222,7 @@ module Make (Current : Version.S) (K : Irmin.Type.S) (V : Irmin.Hash.S) = struct
 
   let list t =
     Log.debug (fun l -> l "[branches] list");
+    if IO.readonly t.block then sync_offset t;
```

Comment:
Why is this needed? Is it because we don't have a `sync` for atomic-write stores? I'm not sure why this is related to the ordering in which we open the RO/RW instances?